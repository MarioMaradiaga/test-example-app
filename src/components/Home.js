import { useState, useCallback } from "react";

function Home() {
  const [showDescription, setShowDescription] = useState(false);
  const onClick = useCallback(() => {
    setShowDescription(true);
  }, []);

  return (
    <div className="home">
      <h1>Home</h1>
      <button onClick={onClick}>Click me</button>
      {showDescription && (
        <div className="description">
          Hey there, I'm some behavioural functionality
        </div>
      )}
    </div>
  );
}

export default Home;
