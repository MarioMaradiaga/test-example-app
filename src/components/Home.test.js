import Home from "./Home";
import { shallow } from "enzyme";

import renderer from "react-test-renderer";

test("should render Home page", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render description on button click", () => {
  const mockCallBack = jest.fn();

  const HomePage = shallow(<Home />);
  HomePage.find("button").simulate("click");
  expect(HomePage.find("[className='description']").length).toEqual(1);
});
