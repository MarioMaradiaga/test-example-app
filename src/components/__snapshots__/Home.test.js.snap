// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home page 1`] = `
<div
  className="home"
>
  <h1>
    Home
  </h1>
  <button
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
